master/buildbot/statistics/capture.py

```diff
@@ -0,0 +1,200 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from twisted.internet import defer
+from twisted.internet import threads
+
+
+class Capture(object):
+    """
+    Base class for all Capture* classes.
+    """
+    def __init__(self, routingKey, callback):
+        self.routingKey = routingKey
+        self.callback = callback
+        # parent service and buildmaster to be set when StatsService initialized
+        self.parent_svcs = []
+        self.master = None
+
+    def defaultContext(self, msg):
+        return {
+            "builder_name": self.builder_name,
+            "build_number": str(msg['number'])
+        }
+
+    def consumer(self, routingKey, msg):
+        raise NotImplementedError
+
+
+class CaptureProperty(Capture):
+    """
+    Convenience wrapper for getting statistics for filtering.
+    Filters out build properties specifies in the config file.
+    """
+    def __init__(self, builder_name, property_name, callback=None):
+        self.builder_name = builder_name
+        self.property_name = property_name
+        routingKey = ("builders", None, "builds", None, "finished")
+
+        def default_callback(props, property_name):
+            return props[property_name][0]  # index: 0 - prop_value, 1 - prop_source
```

Comment:
I'm not sure. The user should be made aware that the there's a problem with the build. If we catch the error here and not send the data over to InfluxDB, the user might not know there a problem with their build for a long time, even with all the logging. This might cause them to miss build data of several builds. I'll discuss it with the mentors tomorrow (Friday).